### PR TITLE
helm: remove topology RBAC section for cephfs deployment

### PR DESCRIPTION
**Pull request by humblec**
At present CephFS does not support topology aware provisioning and this
commit remove the same.

Updates: #3018

### CODE REVIEW

File: charts/ceph-csi-cephfs/templates/provisioner-clusterrole.yaml [hunk 1]
```diff
@@ -53,12 +53,4 @@ rules:
     resources: ["persistentvolumeclaims/status"]
     verbs: ["update", "patch"]
 {{- end -}}
-{{- if .Values.topology.enabled }}
```
Comment (Madhu-1) [inline charts/ceph-csi-cephfs/templates/provisioner-clusterrole.yaml:56]: please remove the topology section in value.yaml also?

Comment (humblec) [reply charts/ceph-csi-cephfs/templates/provisioner-clusterrole.yaml:56]: @Madhu-1 sure , done

File: charts/ceph-csi-cephfs/values.yaml [hunk 2]
```diff
@@ -206,18 +206,6 @@ provisioner:
...
-  # Specifies whether topology based provisioning support should
-  # be exposed by CSI
-  enabled: false
-  # domainLabels define which node labels to use as domains
-  # for CSI nodeplugins to advertise their domains
-  # NOTE: the value here serves as an example and needs to be
-  # updated with node labels that define domains of interest
-  domainLabels:
-    - failure-domain/region
-    - failure-domain/zone
```
Comment (Madhu-1) [inline charts/ceph-csi-cephfs/values.yaml:219]: Please update Readme also as these values are getting removed.

**Review by Madhu-1 — CHANGES_REQUESTED** [review]

File: charts/ceph-csi-cephfs/values.yaml (hunk 2 above)
Comment (humblec) [reply charts/ceph-csi-cephfs/values.yaml:219]: @Madhu-1 where exactly you refer

Comment (Madhu-1) [reply charts/ceph-csi-cephfs/values.yaml:219]: @humblec here https://github.com/ceph/ceph-csi/tree/devel/charts/ceph-csi-cephfs#configuration

Comment (humblec) [reply charts/ceph-csi-cephfs/values.yaml:219]: @Madhu-1 updated.thanks

**Review by Madhu-1 — COMMENTED** [review]
changes LGTM, @humblec do we need 2 commits for this one?

**Comment by humblec** [comment]
> changes LGTM, @humblec do we need 2 commits for this one?

@Madhu-1  I am not sure either.. I am kind of lost on the arrangement of commits in this repo atleast :( . most of the time I was asked to keep it in different commits if the changes are at different code areas. so kept it in different .

**Comment by Madhu-1** [comment]
> > changes LGTM, @humblec do we need 2 commits for this one?
> 
> @Madhu-1 I am not sure either.. I am kind of lost on the arrangement of commits in this repo atleast :( . most of the time I was asked to keep it in different commits if the changes are at different code areas. so kept it in different .

Even am also, don't like different commits one for doc, one for the helm, and deployment builde.env,e2e  for the same change. this might cause issues in cherry-pick etc. lets discuss this in tomorrow's standup.

**Review by Madhu-1 — APPROVED** [review]
Approving this one for now as this pattern is followed and needs to be sorted out soon.

**Comment by humblec** [comment]
@yati1998 ptal.. thanks.

**Comment by nixpanic** [comment]
> Even am also, don't like different commits one for doc, one for the helm, and deployment builde.env,e2e for the same change. this might cause issues in cherry-pick etc. lets discuss this in tomorrow's standup.

My preference is to have commits address a single thing. In this case both commits are part of a small cleanup, which can definitely go together.

For large PRs, that introduce new features, helper function and their unit-tests can go together, consumption of the new functions (if complex enough) should probably go into its own commit, just like documentation with examples.

Splitting commits is done to help maintainers and contributors reviewing the PR and it updates, as well as checking the history of code while working on it at a later time.

**Review by nixpanic — APPROVED** [review]

**Pull request merged**